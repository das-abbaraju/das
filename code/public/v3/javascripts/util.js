/* This 'util' module will hold our filters. */
var util = angular.module('util', [])

/* The 'gender' filter. */
util.filter('gender', function() {
    return function(input) {
        switch(input) {
            case "M": return "male";
            case "F": return "female";
            default:  return "unknown";
        }
    }
});
