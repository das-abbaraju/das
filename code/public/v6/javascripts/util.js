/* This 'util' module will hold our filters. */
var util = angular.module('util', [])

/* The 'gender' filter. */
util.filter('gender', function() {
    return function(input) {
        switch(input) {
            case "M": return "male";
            case "F": return "female";
            default:  return "unknown";
        }
    }
});


util.factory('picsnotify', function($rootScope, $timeout) {
    $rootScope.warning = null;
    $rootScope.error   = null;

    return {
        showWarning: function(msg) {
            if (msg) {
                $rootScope.warning = msg;
                $timeout(function() { $rootScope.warning = null }, 3000);
            }
        },
        showError: function(msg) {
            if (msg) {
                $rootScope.error = msg;
                $timeout(function() { $rootScope.error = null }, 3000);
            }
        }
    }
});

util.factory('picsappHttp', function($http, picsnotify) {
    function handleError(errorData) {
        picsnotify.showError(errorData.error);
    }

    return {
        get: function(url, successCallback) {
            $http.get(url).success(successCallback).error(handleError);
        },
        post: function(url, data, successCallback) {
            $http.post(url, data).success(successCallback).error(handleError);
        }
    }
});
