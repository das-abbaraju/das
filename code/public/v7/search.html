<div>
    <div class="row">
        <div class="col-md-6">
            <form action="" id="search-form">
                <span class="glyphicon glyphicon-search"></span>
                <input type="text" size="50"
                       placeholder="Enter the first part of a last name"
                       ng-change="checkSearch()"
                       ng-model="searchPrefix"/>
            </form>
        </div>
    </div>

    <div id="search-results" class="row">
        <div class="col-md-12">
            <div ng-hide="people">No search results.</div>

            <div ng-show="people">{{ people.length }} matching people</div>

            <table class="table" ng-show="people" id="table-template">
                <thead>
                <tr>
                    <th>
                        <a href="" ng-click="toggleSortOrder('last')">Name</a>
                        <span ng-show="!reverse && (predicate == 'last')">&#x25b2;</span>
                        <span ng-show="reverse && (predicate == 'last')">&#x25bc;</span>
                    </th>

                    <th>
                        <a href="" ng-click="toggleSortOrder('ssn')">SSN</a>
                        <span ng-show="!reverse && (predicate == 'ssn')">&#x25b2;</span>
                        <span ng-show="reverse && (predicate == 'ssn')">&#x25bc;</span>
                    </th>

                    <th>
                        <a href="" ng-click="toggleSortOrder('birthDate')">Birth Date</a>
                  <span ng-show="!reverse && (predicate == 'birthDate')">&#x25b2;</sp  an>
                  <span ng-show="reverse && (predicate == 'birthDate')">&#x25bc;</spa  n>
                    </th>

                    <th>
                        <a href="" ng-click="toggleSortOrder('gender')">Gender</a>
                        <span ng-show="!reverse && (predicate == 'gender')">&#x25b2;</span>
                        <span ng-show="reverse && (predicate == 'gender')">&#x25bc;</span>
                    </th>
                </tr>
                </thead>

                <tbody>
                <tr id="template-row"
                    ng-repeat="person in people | orderBy:predicate:reverse">
                    <td class="name">
                        <a ng-href="#/show/{{person.id}}">{{person.name}}</a>
                    </td>
                    <td class="SSN">{{person.ssn}}</td>

                    <td class="DOB">{{person.birthDate | date:'d MMM, yyyy'}}</td>
                    <td class="gender">{{person.gender | gender}}</td>
                </tr>
                </tbody>
            </table>
        </div>
    </div>
</div> <!-- search -->
