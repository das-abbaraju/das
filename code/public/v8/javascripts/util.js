/* This 'util' module will hold our filters. */
var util = angular.module('util', [])

/* The 'gender' filter. */
util.filter('gender', function() {
    return function(input) {
        switch(input) {
            case "M": return "male";
            case "F": return "female";
            default:  return "unknown";
        }
    }
});


util.directive('messageBox', function() {
    return {
        restrict: 'E',
        scope: {
            ngModel: '=',
            type: '@',
            icon: '@'
        },
        transclude: true,
        templateUrl: 'templates/message-box.html',

        link: function (scope, element, attrs, $timeout) {
            scope.close = function () {
                scope.ngModel = null;
            }

        }
    }// The meat of the directive goes below.
});

util.factory('picsnotify', function($rootScope, $timeout) {
    $rootScope.warning = null;
    $rootScope.error   = null;

    return {
        showWarning: function(msg) {
            if (msg) {
                $rootScope.warning = msg;
                $timeout(function() { $rootScope.warning = null }, 3000);
            }
        },
        showError: function(msg) {
            if (msg) {
                $rootScope.error = msg;
                $timeout(function() { $rootScope.error = null }, 3000);
            }
        }
    }
});

util.factory('picsappHttp', function($http, picsnotify) {
    function handleError(errorData) {
        picsnotify.showError(errorData.error);
    }

    return {
        get: function(url, successCallback) {
            $http.get(url).success(successCallback).error(handleError);
        },
        post: function(url, data, successCallback) {
            $http.post(url, data).success(successCallback).error(handleError);
        }
    }
});
