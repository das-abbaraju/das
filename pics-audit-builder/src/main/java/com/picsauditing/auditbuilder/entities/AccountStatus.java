package com.picsauditing.auditbuilder.entities;

public enum AccountStatus {
    Active,
	Pending,
	Requested,
	Demo,
	Deleted,
	Deactivated,
	Declined;

    public boolean isDeleted() {
        return this == Deleted;
    }

    public boolean isDeactivated() {
        return this == Deactivated;
    }
}