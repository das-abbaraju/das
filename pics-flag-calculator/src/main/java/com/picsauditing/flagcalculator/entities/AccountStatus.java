package com.picsauditing.flagcalculator.entities;

public enum AccountStatus {
    Active/*(false)*/,
    Pending/*(true)*/,
    Requested/*(true)*/,
    Demo/*(true)*/,
    Deleted/*(true)*/,
    Deactivated/*(true)*/,
    Declined/*(true)*/;

//    private boolean enforcePermissions;
//
//    AccountStatus(boolean enforcePermissions) {
//        this.enforcePermissions = enforcePermissions;
//    }
//
//    public boolean isRequested() {
//        return this == Requested;
//    }
//
//    public boolean isActive() {
//        return this == Active;
//    }
//
//    public boolean isPending() {
//        return this == Pending;
//    }
//
//    public boolean isDemo() {
//        return this == Demo;
//    }
//
//    public boolean isDeleted() {
//        return this == Deleted;
//    }
//
//    public boolean isDeactivated() {
//        return this == Deactivated;
//    }
//
//    public boolean isDeclinedDeletedDeactivated() {
//        return this == Deleted || this == Deactivated || this == Declined;
//    }
//
//    @Deprecated
//    // TODO find any usages within JSPs and rename them
//    public boolean isPendingDeactivated() {
//        return isPendingOrDeactivated();
//    }
//
//    public boolean isPendingOrDeactivated() {
//        return this == Pending || this == Deactivated;
//    }
//
//    public boolean isPendingRequestedOrDeactivated() {
//        return this == Pending || this == Deactivated || this == Requested;
//    }
//
//    public boolean isPendingRequestedDeclinedOrDeactivated() {
//        return this == Pending || this == Deactivated || this == Requested || this == Declined;
//    }
//
//    public boolean isPendingDeclinedOrDemo() {
//        return this == Pending || this == Declined || this == Demo;
//    }
//
//    @Deprecated
//    // TODO find any usages within JSPs and rename them
//    public boolean isActiveDemo() {
//        return isActiveOrDemo();
//    }
//
//    public boolean isActiveOrDemo() {
//        return this == Active || this == Demo;
//    }
//
//
//    public boolean isDeclined() {
//        return this == Declined;
//    }
//
//    @Transient
//    @Override
//    public String getI18nKey() {
//        return this.getClass().getSimpleName() + "." + this.name();
//    }
//
//    @Transient
//    @Override
//    public String getI18nKey(String property) {
//        return getI18nKey() + "." + property;
//    }
//
//    public boolean isVisibleTo(Permissions permissions) {
//        if (enforcePermissions)
//            return permissions.hasPermission(OpPerms.AllContractors);
//
//        return true;
//    }
//
//    public boolean isActivePending() {
//        return this == Active || this == Pending;
//    }
//
//    public boolean isActiveDemoPending() {
//        return this == Active || this == Pending || this == Demo;
//    }
//
//    public boolean isActivePendingRequested() {
//        return this == Active || this == Pending || this == Requested;
//    }
//
//    public boolean allowRegistrationToBeMarkedDeclined() {
//        return this == Pending || this == Requested;
//    }
//
//    public boolean allowRegistrationToBeMarkedDuplicated() {
//        return this == Pending || this == Requested;
//    }
}
