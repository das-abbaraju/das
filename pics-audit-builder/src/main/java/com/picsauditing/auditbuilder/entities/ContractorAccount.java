package com.picsauditing.auditbuilder.entities;

import org.hibernate.annotations.Sort;
import org.hibernate.annotations.SortType;
import org.hibernate.annotations.Where;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;
import java.util.Set;
import java.util.TreeSet;

@SuppressWarnings("serial")
@Entity(name = "com.picsauditing.auditbuilder.entities.ContractorAccount")
@Table(name = "contractor_info")
public class ContractorAccount extends Account {

    private boolean safetySensitive;
	private LowMedHigh safetyRisk = LowMedHigh.None;
	private LowMedHigh productRisk = LowMedHigh.None;
    private LowMedHigh transportationRisk = LowMedHigh.None;
	private boolean soleProprietor;
	private AccountLevel accountLevel = AccountLevel.Full;

	private List<ContractorAudit> audits = new ArrayList<>();
	private List<ContractorOperator> operators = new ArrayList<>();
	private Set<ContractorTrade> trades = new TreeSet<>();

	@OneToMany(mappedBy = "contractorAccount", cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH})
	@Where(clause = "expiresDate > NOW() OR expiresDate IS NULL")
	public List<ContractorAudit> getAudits() {
		return this.audits;
	}

	public void setAudits(List<ContractorAudit> audits) {
		this.audits = audits;
	}

	@OneToMany(mappedBy = "contractorAccount", cascade = {CascadeType.REMOVE, CascadeType.REFRESH})
	public List<ContractorOperator> getOperators() {
		return this.operators;
	}

	public void setOperators(List<ContractorOperator> operators) {
		this.operators = operators;
	}

    @Column(name = "safetySensitive", nullable = false)
    public boolean isSafetySensitive() {
        return safetySensitive;
    }

    public void setSafetySensitive(boolean safetySensitive) {
        this.safetySensitive = safetySensitive;
    }

    @Column(name = "safetyRisk", nullable = false)
	public LowMedHigh getSafetyRisk() {
		return safetyRisk;
	}

	public void setSafetyRisk(LowMedHigh safetyRisk) {
		this.safetyRisk = safetyRisk;
	}

	@Column(name = "productRisk", nullable = false)
	public LowMedHigh getProductRisk() {
		return productRisk;
	}

	public void setProductRisk(LowMedHigh productRisk) {
		this.productRisk = productRisk;
	}

    @Column(name = "transportationRisk", nullable = false)
    public LowMedHigh getTransportationRisk() {
        return transportationRisk;
    }

    public void setTransportationRisk(LowMedHigh transportationRisk) {
        this.transportationRisk = transportationRisk;
    }

	@OneToMany(mappedBy = "contractor")
	public Set<ContractorTrade> getTrades() {
		return trades;
	}

	public void setTrades(Set<ContractorTrade> trades) {
		this.trades = trades;
	}

	public boolean getSoleProprietor() {
		return soleProprietor;
	}

	public void setSoleProprietor(boolean soleProprietor) {
		this.soleProprietor = soleProprietor;
	}

	@Enumerated(EnumType.STRING)
	public AccountLevel getAccountLevel() {
		return accountLevel;
	}

	public void setAccountLevel(AccountLevel accountLevel) {
		this.accountLevel = accountLevel;
	}
}