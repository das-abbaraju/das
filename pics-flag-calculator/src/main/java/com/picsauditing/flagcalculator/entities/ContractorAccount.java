package com.picsauditing.flagcalculator.entities;

import org.hibernate.annotations.Cache;
import org.hibernate.annotations.*;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.*;
import javax.persistence.Table;
import java.util.*;

@SuppressWarnings("serial")
@Entity(name = "com.picsauditing.flagcalculator.entities.ContractorAccount")
@Table(name = "contractor_info")
// Cache is only on the operator account now, if this works.
@Cache(usage = CacheConcurrencyStrategy.READ_WRITE, region = "daily")
public class ContractorAccount extends Account {
    private boolean soleProprietor;
    private AccountLevel accountLevel = AccountLevel.Full;
    private Set<FlagCriteriaContractor> flagCriteria = new HashSet<>();
    private List<ContractorAudit> audits = new ArrayList<ContractorAudit>();
    private List<ContractorOperator> operators = new ArrayList<>();
    private List<ContractorTag> operatorTags = new ArrayList<ContractorTag>();
    private Set<ContractorTrade> trades = new HashSet<>();

    @OneToMany(mappedBy = "contractorAccount", cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH})
    @Where(clause = "expiresDate > NOW() OR expiresDate IS NULL")
    public List<ContractorAudit> getAudits() {
        return this.audits;
    }

    public void setAudits(List<ContractorAudit> audits) {
        this.audits = audits;
    }

    @OneToMany(mappedBy = "contractorAccount", cascade = {CascadeType.REMOVE, CascadeType.REFRESH})
    public List<ContractorOperator> getOperators() {
        return this.operators;
    }

    public void setOperators(List<ContractorOperator> operators) {
        this.operators = operators;
    }

    @OneToMany(mappedBy = "contractor", cascade = {CascadeType.REMOVE, CascadeType.MERGE})
    public List<ContractorTag> getOperatorTags() {
        return operatorTags;
    }

    public void setOperatorTags(List<ContractorTag> operatorTags) {
        this.operatorTags = operatorTags;
    }

    @OneToMany(mappedBy = "contractor")
    public Set<ContractorTrade> getTrades() {
        return trades;
    }

    public void setTrades(Set<ContractorTrade> trades) {
        this.trades = trades;
    }

    @OneToMany(mappedBy = "contractor", cascade = {CascadeType.ALL})
    public Set<FlagCriteriaContractor> getFlagCriteria() {
        return flagCriteria;
    }

    public void setFlagCriteria(Set<FlagCriteriaContractor> flagCriteria) {
        this.flagCriteria = flagCriteria;
    }

    public boolean getSoleProprietor() {
        return soleProprietor;
    }

    public void setSoleProprietor(boolean soleProprietor) {
        this.soleProprietor = soleProprietor;
    }

    @Enumerated(EnumType.STRING)
    public AccountLevel getAccountLevel() {
        return accountLevel;
    }

    public void setAccountLevel(AccountLevel accountLevel) {
        this.accountLevel = accountLevel;
    }
}
