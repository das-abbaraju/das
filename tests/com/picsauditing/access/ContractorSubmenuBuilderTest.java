package com.picsauditing.access;

import com.picsauditing.PICS.I18nCache;
import com.picsauditing.dao.ContractorAccountDAO;
import com.picsauditing.jpa.entities.AccountStatus;
import com.picsauditing.jpa.entities.ContractorAccount;
import com.picsauditing.report.ReportUtil;
import com.picsauditing.search.Database;
import org.approvaltests.Approvals;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.json.simple.JSONArray;
import org.junit.*;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;
import org.mockito.invocation.InvocationOnMock;
import org.mockito.stubbing.Answer;
import org.powermock.reflect.Whitebox;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Locale;

import static org.mockito.Matchers.any;
import static org.mockito.Matchers.anyString;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

@UseReporter(DiffReporter.class)
public class ContractorSubmenuBuilderTest {

	public static final int ACCOUNT_ID = 5;

	private ContractorSubmenuBuilder contractorSubmenuBuilder;
	@Mock
	private Permissions permissions;
	@Mock
	protected I18nCache i18nCache;
	@Mock
	private ContractorAccount contractorAccount;
	@Mock
	private ContractorAccountDAO contractorAccountDao;

	@BeforeClass
	public static void setUpClass() throws Exception {
		Whitebox.setInternalState(I18nCache.class, "databaseForTesting", mock(Database.class));
	}

	@Before
	public void setUp() throws Exception {
		contractorSubmenuBuilder = new ContractorSubmenuBuilder();
		MockitoAnnotations.initMocks(this);

		when(permissions.getAccountId()).thenReturn(ACCOUNT_ID);
		when(i18nCache.getText(anyString(), any(Locale.class))).then(returnMockTranslation());
		Whitebox.setInternalState(ReportUtil.class, "i18nCache", i18nCache);
	}

	@AfterClass
	public static void tearDownClass() {
		Whitebox.setInternalState(I18nCache.class, "databaseForTesting", (Database) null);
		Whitebox.setInternalState(ReportUtil.class, "i18nCache", (I18nCache) null);
	}

	@Test
	public void testAddCompanyMenu_defaultCase() throws Exception {
		MenuComponent menubar = new MenuComponent();

		menubar = contractorSubmenuBuilder.addCompanyMenu(menubar, permissions);

		JSONArray result = MenuWriter.convertMenuToJSON(menubar);
		Approvals.verify(result.toString());
	}

	@Test
	public void testAddCompanyMenu_admin() throws Exception {
		when(permissions.isAdmin()).thenReturn(true);
		MenuComponent menubar = new MenuComponent();

		menubar = contractorSubmenuBuilder.addCompanyMenu(menubar, permissions);

		JSONArray result = MenuWriter.convertMenuToJSON(menubar);
		Approvals.verify(result.toString());
	}

	@Test
	public void testAddCompanyMenu_adminAndDemo() throws Exception {
		when(permissions.isAdmin()).thenReturn(true);
		when(permissions.getAccountStatus()).thenReturn(AccountStatus.Demo);
		MenuComponent menubar = new MenuComponent();

		menubar = contractorSubmenuBuilder.addCompanyMenu(menubar, permissions);

		JSONArray result = MenuWriter.convertMenuToJSON(menubar);
		Approvals.verify(result.toString());
	}

	@Test
	public void testAddCompanyMenu_contractor() throws Exception {
		when(permissions.isContractor()).thenReturn(true);
		MenuComponent menubar = new MenuComponent();

		menubar = contractorSubmenuBuilder.addCompanyMenu(menubar, permissions);

		JSONArray result = MenuWriter.convertMenuToJSON(menubar);
		Approvals.verify(result.toString());
	}

	@Test
	public void testAddCompanyMenu_generalContractor() throws Exception {
		when(permissions.isGeneralContractor()).thenReturn(true);
		MenuComponent menubar = new MenuComponent();

		menubar = contractorSubmenuBuilder.addCompanyMenu(menubar, permissions);

		JSONArray result = MenuWriter.convertMenuToJSON(menubar);
		Approvals.verify(result.toString());
	}

	@Test
	public void testAddDocuguardMenu_defaultCase() throws Exception {
		MenuComponent menubar = new MenuComponent();
		List<MenuComponent> auditMenu = buildTestAuditMenu();

		menubar = contractorSubmenuBuilder.addDocuguardMenu(menubar, permissions, auditMenu);

		JSONArray result = MenuWriter.convertMenuToJSON(menubar);
		Approvals.verify(result.toString());
	}

	@Test
	public void testAddAuditguardMenu_defaultCase() throws Exception {
		MenuComponent menubar = new MenuComponent();
		List<MenuComponent> auditMenu = buildTestAuditMenu();

		menubar = contractorSubmenuBuilder.addAuditguardMenu(menubar, permissions, auditMenu);

		JSONArray result = MenuWriter.convertMenuToJSON(menubar);
		Approvals.verify(result.toString());
	}

	@Test
	public void testAddInsureguardMenu_defaultCase() throws Exception {
		MenuComponent menubar = new MenuComponent();
		List<MenuComponent> auditMenu = buildTestAuditMenu();

		menubar = contractorSubmenuBuilder.addInsureguardMenu(menubar, permissions, auditMenu);

		JSONArray result = MenuWriter.convertMenuToJSON(menubar);
		Approvals.verify(result.toString());
	}

	@Test
	public void testAddSupportMenu_defaultCase() throws Exception {
		MenuComponent menubar = new MenuComponent();

		menubar = contractorSubmenuBuilder.addSupportMenu(menubar, permissions, true);

		JSONArray result = MenuWriter.convertMenuToJSON(menubar);
		Approvals.verify(result.toString());
	}

	@Test
	public void testAddEmployeeguardMenu_defaultCase() throws Exception {
		when(contractorAccount.isHasEmployeeGUARDTag()).thenReturn(true);
		when(permissions.isContractor()).thenReturn(false);
		when(permissions.isAdmin()).thenReturn(true);
		MenuComponent menubar = new MenuComponent();

		menubar = contractorSubmenuBuilder.addEmployeeguardMenu(menubar, permissions, contractorAccount);

		JSONArray result = MenuWriter.convertMenuToJSON(menubar);
		Approvals.verify(result.toString());
	}

	private List<MenuComponent> buildTestAuditMenu() {
		List<MenuComponent> auditMenu = new ArrayList<>();

		// PQF
		MenuComponent pqfComponent = new MenuComponent(ContractorSubmenuBuilder.PQF, "/ContractorDocuments.action?id=3");

		MenuComponent childItem = new MenuComponent(ContractorSubmenuBuilder.PQF, "/Audit.action?auditID=1");
		childItem.setAuditId(10);
		pqfComponent.getChildren().add(childItem);

		auditMenu.add(pqfComponent);

		// Auditguard
		MenuComponent auditguardComponent = new MenuComponent(ContractorSubmenuBuilder.AUDITGUARD, "/ContractorDocuments.action?id=3");

		childItem = new MenuComponent(ContractorSubmenuBuilder.MANUAL_AUDIT + " '12", "/Audit.action?auditID=1");
		childItem.setAuditId(20);
		auditguardComponent.getChildren().add(childItem);

		childItem = new MenuComponent(ContractorSubmenuBuilder.IMPLEMENTATION_AUDIT + " '12", "/Audit.action?auditID=1");
		childItem.setAuditId(30);
		auditguardComponent.getChildren().add(childItem);

		auditMenu.add(auditguardComponent);

		// Insureguard
		MenuComponent insureguardComponent = new MenuComponent(ContractorSubmenuBuilder.INSUREGUARD, "/ContractorDocuments.action?id=3");

		childItem = new MenuComponent(ContractorSubmenuBuilder.MANAGE_CERTIFICATES, "/Audit.action?auditID=1");
		childItem.setAuditId(40);
		insureguardComponent.getChildren().add(childItem);

		childItem = new MenuComponent(ContractorSubmenuBuilder.AUTOMOBILE_LIABILITY + " (New)", "/Audit.action?auditID=1");
		childItem.setAuditId(50);
		insureguardComponent.getChildren().add(childItem);

		childItem = new MenuComponent(ContractorSubmenuBuilder.EXCESS_UMBRELLA_LIABILITY + " (New)", "/Audit.action?auditID=1");
		childItem.setAuditId(60);
		insureguardComponent.getChildren().add(childItem);

		childItem = new MenuComponent(ContractorSubmenuBuilder.GENERAL_LIABILITY + " (New)", "/Audit.action?auditID=1");
		childItem.setAuditId(70);
		insureguardComponent.getChildren().add(childItem);

		childItem = new MenuComponent(ContractorSubmenuBuilder.WORKER_COMP + " (New)", "/Audit.action?auditID=1");
		childItem.setAuditId(80);
		insureguardComponent.getChildren().add(childItem);

		auditMenu.add(insureguardComponent);

		return auditMenu;
	}

	private Answer<String> returnMockTranslation() {
		return new Answer<String>() {
			@Override
			public String answer(InvocationOnMock invocation) throws Throwable {
				Object[] args = invocation.getArguments();
				return Arrays.toString(args);
			}
		};
	}

}
