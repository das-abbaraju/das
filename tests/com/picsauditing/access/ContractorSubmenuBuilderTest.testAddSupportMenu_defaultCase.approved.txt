[{"text":"menu.Support","menu":{"items":[{"id":"help_center_menu","text":"Header.HelpCenter","href":"http:\/\/help.picsorganizer.com\/login.action?os_destination=homepage.action&os_username=admin&os_password=ad9870mins"},{"id":"live_chat_menu","text":"Registration.Error.LiveChat","href":"#"},{"id":"contact_action_menu","text":"global.ContactPICS","href":"Contact.action"},{"id":"about_pics_menu","text":"global.AboutPICS","href":"About.action"},{"text":"Reference","menu":{"items":[{"id":"navigation_menu_menu","text":"Navigation Menu","href":"Reference!navigationMenu.action"},{"id":"dynamic_report_menu","text":"Dynamic Reports","href":"Reference!dynamicReport.action"},{"id":"reports_manager_menu","text":"Reports Manager","href":"Reference!reportsManager.action"}]}}]}}]